src/garage/envs/multi_env_wrapper.py

```diff
@@ -187,7 +177,7 @@ def reset(self):
         if self._mode == 'vanilla':
             pass
         elif self._mode == 'add-onehot':
-            obs = np.concatenate([obs, self._active_task_one_hot()])
+            episode_info['task_one_hot'] = self._active_task_one_hot()
         else:  # self._mode == 'del-onehot'
             obs = obs[:-self._num_tasks]
```

Comment:
I'd suggest to remove `mode` from MultiEnvWrapper in this PR. The active task index should *always* be in `episode_info`, we don't need a mode to determine that. 
Once you done that you can remove the if conditions here and simply do `episode_info['task_one_hot'] = self._active_task_one_hot()`.